russh/src/client/mod.rs

```diff
@@ -723,6 +723,16 @@ async fn start_reading<R: AsyncRead + Unpin>(
     Ok((n, stream_read, buffer, cipher))
 }
 
+fn future_or_pending<F: futures::Future, T>(
```

Comment:
Could this function be avoided by using `.is_some()` as a precondition in `tokio::select!`?

Reply:
I'm afraid not. An `Option<impl Future>` doesn't also implement `Future`, and even if the precondition disables a branch, it still always evaluates the async expression and expects it to be the same type each time. So there has to be _something_ pollable to put in the place of an absent `Sleep`, and the `Pending` future is the obvious choice since it will efficiently never complete.

That said, this function has only two callsites, and could be inlined back into those callsites - I only factored it out for DRY purposes.